### PR TITLE
sbt 1.2

### LINKED ISSUE #3192: Upgrade to sbt 1.0

### PR DESCRIPTION
**Pull request by jstriebel**
This depends on #3251.

Changes include:
- upgrade to sbt 1.2.3
- removal of unused sbt plugins & resolvers
- using coursier for dependency downloads, adapted caches for that
- refactor build.sbt

:warning: When running this locally, also run `docker-compose build compile`

### URL of deployed dev instance (used for testing):
- https://sbtoneforked.webknossos.xyz

### Steps to test:
- CI should catch most things
- testing most important things in dev-instance
- double check buildinfo

### Issues:
- fixes #3192

------
- [ ] ~~Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- [ ] ~~Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- [ ] ~~Updated [documentation](../blob/master/docs) if applicable~~
- [x] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -23,7 +23,7 @@ lazy val webknossosSettings = Seq(
   libraryDependencies ++= Dependencies.webknossosDependencies,
   resolvers ++= DependencyResolvers.dependencyResolvers,
   sourceDirectory in Assets := file("none"),
-  updateOptions := updateOptions.value.withLatestSnapshots(true),
+  updateOptions := updateOptions.value.withLatestSnapshots(true).withCachedResolution(true),
```
Comment (jstriebel) [inline build.sbt]: this change might be useless, I should test this in the end

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/TemporaryStore.scala [hunk 2]
```diff
@@ -1,36 +1,51 @@
...
 import javax.inject.Inject
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.stm.Ref
 
 class TemporaryStore[K, V] @Inject()(system: ActorSystem) {
 
-  lazy val ts = Agent[Map[K, V]](Map())
+  lazy val map: scala.collection.mutable.Map[K, V] = scala.collection.mutable.Map()
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/TemporaryStore.scala]: Not sure if we should include the changes from #3251 here.

File: project/AssetCompilation.scala [hunk 3]
```diff
@@ -113,7 +117,7 @@ object AssetCompilation {
   val settings = Seq(
     AssetCompilation.SettingsKeys.webpackPath := (Path("node_modules") / ".bin" / "webpack").getPath,
     AssetCompilation.SettingsKeys.npmPath := "yarn",
-    run in Compile <<= (run in Compile) map(killWebpack) dependsOn webpackGenerateTask,
+    run in Compile := {(run in Compile) map(killWebpack) dependsOn webpackGenerateTask},
```
Comment (jstriebel) [inline project/AssetCompilation.scala]: This currently leads to `sbt run` not working. Tbh, I'm not sure what the previous semantic was. Any ideas here, @fm3 ?
https://www.scala-sbt.org/1.x/docs/Howto-After-Input-Task.html

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/TemporaryStore.scala (hunk 2 above)
Comment (fm3) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/TemporaryStore.scala]: We should merge #3251 first, but I think we can do that first thing monday morning :)

File: project/AssetCompilation.scala (hunk 3 above)
Comment (fm3) [reply project/AssetCompilation.scala]: Which version do you mean breaks sbt run? the old `<<=` or the new `:=`? I’m not sure about the different operators here, either. But I think the intended semantic of this line was that `run` should execute `webpackGenerateTask` before itself and `killWebpack` after itself

Comment (jstriebel) [reply project/AssetCompilation.scala]: @rschwanhold already figured out the correct syntax in #3251. Let's merge #3251 first, then I'll merge it in here.

**Comment by jstriebel** [comment]
@fm3 For the verbose e2e-tests: I don't see how that should come from the scala changes. My best guess that this is because of a new node or yarn version, that comes with the rebuild docker-image: 
https://github.com/scalableminds/webknossos/blob/master/Dockerfile.dev#L10-L20

@philippotto @daniel-wer Do you have an idea, how to suppress the superfluous output from e2e-tests? It seems to arise from a new node or yarn version. It looks like this:
```
Resetting test database...
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
Fetching http://localhost:9000/api/datasets
  ✔ backend-snapshot-tests › datasets.e2e › getDatasets (3s)
```

**Comment by philippotto** [comment]
@jstriebel I think @fm3 was talking about some other output ("thousands of lines"). The output you pasted above is "normal" (as in: "has been like that before"). @fm3 and I think that we can leave the output as is and if we notice something weird in the future, we can re-evaluate.

**Comment by jstriebel** [comment]
@fm3 Is anything missing here? If not we can merge and deploy this on Thursday.

**Pull request merged**